src/bin/mcp.ts

```diff
@@ -22,19 +22,55 @@ This is a running process of the Firebase MCP server. This command should only b
 }
 `;
 
+const HELP_TEXT = `Usage: firebase mcp [options]
+
+Description:
+  Starts the Model Context Protocol (MCP) server for the Firebase CLI. This server provides a
+  standardized way for AI agents and IDEs to interact with your Firebase project.
+
+Tool Discovery & Loading:
+  The server automatically determines which tools to expose based on your project context.
+
+  1. Auto-Detection (Default):
+     - Scans 'firebase.json' for configured services (e.g., Hosting, Firestore).
+     - Checks enabled Google Cloud APIs for the active project.
+     - Inspects project files for specific SDKs (e.g., Crashlytics in Android/iOS).
+
+  2. Manual Overrides:
+     - Use '--only' to restrict tool discovery to specific feature sets (e.g., core, firestore).
+     - Use '--tools' to disable auto-detection entirely and load specific tools by name.
+
+Options:
+  --dir <path>              Project root directory (defaults to current working directory).
+  --only <features>         Comma-separated list of features to enable (e.g. core, firestore).
+                            If specified, auto-detection is disabled for other features.
+  --tools <tools>           Comma-separated list of specific tools to enable. Disables
+                            auto-detection entirely.
+  --generate-tool-list      Generate markdown documentation for all available tools.
```

Comment:
We can hide these `--generate-*` flags - we've been using them to generate our internal docs, but they probably aren't useful for external users.